projects/plugins/jetpack/extensions/blocks/conversation/utils.js

```diff
@@ -23,3 +24,171 @@ export function getPlainText( html, escape = false ) {
 
 	return escapeHTML( text );
 }
+
+/**
+ * Return the file extension according to the file name.
+ *
+ * @param {string} filename - file full name.
+ * @returns {string} File extension.
+ */
+export function pickExtensionFromFileName( filename ) {
+	return `.${ filename.substr( filename.lastIndexOf( '.' ) + 1 ) }`;
+}
+
+export const FILE_EXTENSION_SRT = '.srt';
+export const FILE_EXTENSION_TXT = '.txt';
+export const FILE_EXTENSION_VTT = '.vtt';
+export const FILE_EXTENSION_SBV = '.sbv';
+
+export const ACCEPTED_FILE_EXT_ARRAY = [
+	FILE_EXTENSION_SRT,
+	FILE_EXTENSION_TXT,
+	FILE_EXTENSION_VTT,
+	FILE_EXTENSION_SBV,
+];
+
+export const ACCEPTED_FILE_EXTENSIONS = ACCEPTED_FILE_EXT_ARRAY.join( ', ' );
+
+export const TRANSCRIPT_MAX_FILE_SIZE = 100000;
+
+/*
+ * Generic format / template.
+ * ----------------------------
+ * <speaker> <timestamp>
+ *  <content>
+ * ----------------------------
+ * Serices: otter.ai, ...
+ */
+const otterFormatRegExp = /(.*[^\s])\s{1,}(\d{1,2}(?::\d{1,2}?)+)\s+\n([\s\S]*?(?=\n{2}|$))/gm;
+const otterFormatTestRegExp = /(.*[^\s])\s{1,}(\d{1,2}(?::\d{1,2}?)+)\s+\n([\s\S]*?(?=\n{2}|$))/g;
+
+/*
+ * ----------------------------
+ * <speaker>: [<timestamp>] <content>
+ * ----------------------------
+ * Serices: sonix.ai, ...
+ */
+const sonixFormatRegExp = /(?:(.*[^\s]):\s*)?\[([\d{1,2}:]*)]\s([\s\S]*?(?=\n{1,2}|$))/gm;
+const sonixFormatTestRegExp = /(.*[^\s]):\s*\[([\d{1,2}:]*)]\s([\s\S]*?(?=\n{1,2}|$))/g;
+
+const parsers = [
+	{
+		name: 'otter',
+		re: otterFormatRegExp,
+		testRE: otterFormatTestRegExp,
+	},
+	{
+		name: 'sonix',
+		re: sonixFormatRegExp,
+		testRE: sonixFormatTestRegExp,
+	},
+];
+
+/* SRT format / template.
+ * ----------------------------
+ * <index>
+ * <startTime> --> <endTime>
+ * <content>
+ * ----------------------------
+ * Serices: otter.ai, youtube.com, etc.
+ */
+export const srtFormatRegExp = /(\d+)\n([\d:,]+)\s+-{2}>\s+([\d:,]+)\n([\s\S]*?(?=\n{2}|$))/gm;
+
+export function isAcceptedTranscriptExtension( fileExtension ) {
+	return ACCEPTED_FILE_EXT_ARRAY.indexOf( fileExtension ) >= 0;
+}
+
+export function SRT_parse( content ) {
+	const result = {
+		conversation: {
+			speakers: [],
+		},
+		dialogues: [],
+	};
+
+	let matches;
+
+	while ( ( matches = srtFormatRegExp.exec( content ) ) !== null ) {
+		result.dialogues.push( {
+			timestamp: matches[ 2 ],
+			content: matches[ 4 ]
+		} );
+	}
+
+	return result;
+}
+
+export function TXT_parse ( content ) {
+	const result = {
+		dialogues: [],
+		conversation: {
+			speakers: [],
+		}
+	};
+
+	// Pick vallid parsers testing the content.
+	const validParsers = parsers.filter( ( { testRE } ) => testRE.test( content ) );
+	if ( ! validParsers?.length ) {
+		return result;
+	}
+
+	// Pick the first parser from the list.
+	const parser = validParsers[ 0 ];
+
+	let matches;
+	while ( ( matches = parser.re.exec( content ) ) != null ) {
```

Comment:
What's the largest file we've tried this with? With one that's 100kb with lots of short dialogue, does this loop (and the resulting insertion of the blocks) lock up rendering of the editor?

Reply:
I've tried with a 75Kb file, getting 132 blocks. It works very well, but definitely, this is something that we'd like to test deeply.

blocks | outline
------|--------
<img alt="Screen Shot 2021-03-04 at 8 23 41 AM" src="https://user-images.githubusercontent.com/77539/109957475-c164ae80-7cc3-11eb-8218-76fde5479e97.png"> | ![image](https://user-images.githubusercontent.com/77539/109957352-911d1000-7cc3-11eb-8f92-a24fd18ba17b.png)

Reply:
Also tested with tiny paragraphs. It takes more time. In my testing, it creates 464 blocks

<img src="https://user-images.githubusercontent.com/77539/109959689-9af44280-7cc6-11eb-830d-2bfdfe942881.png" width="500px" />

Reply:
Cool. It's probably not for this PR anyway, but I was wondering if it made the app unresponsive for any significant length of time.

Reply:
Yeah... this is something that can happen in other scenarios, though. Copy and paste a bunch of content in the canvas for instance could bring similar issues too I guess.